Symfony/CS/Fixer/Contrib/PhpUnitNoDedicateAssertFixer.php

```diff
@@ -0,0 +1,221 @@
+            $assertCallOpen,
+            $testNsToken,
+            $test,
+            $testOpen,
+            $testClose,
+            $assertCallClose,
+        );
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param array  $assertIndexes
+     *
+     * @return int index up till processed, number of tokens added
+     */
+    private function fixAssert(Tokens $tokens, array $assertIndexes)
+    {
+        list(
+            $isPositive,
+            $assertCallIndex,
+            $assertCallOpen,
+            $testNsToken,
+            $test,
+            $testOpen,
+            $testClose,
+            $assertCallClose) = $assertIndexes;
+
+        $content = strtolower($tokens[$test]->getContent());
+        if (array_key_exists($content, $this->fixMap)) {
+            if (false !== $this->fixMap[$content][$isPositive]) {
+                $tokens[$assertCallIndex]->setContent($this->fixMap[$content][$isPositive]);
+                $this->removeFunctionCall($tokens, $testNsToken, $test, $testOpen, $testClose);
+            }
+
+            return $assertCallClose;
+        }
+
+        switch ($content) {
+            case 'is_real':
+                $content = 'is_float';
+                // do not break here
+            case 'is_array':
+            case 'is_bool':
+            case 'is_float':
+            case 'is_int':
+            case 'is_string':
+                $type = substr($content, 3);
+                $tokens[$assertCallIndex]->setContent($isPositive ? 'assertInternalType' : 'assertNotInternalType');
+                $tokens->overrideAt($test, array(T_CONSTANT_ENCAPSED_STRING, "'".$type."'", $tokens[$test]->getLine()));
+                $tokens->overrideAt($testOpen, ',');
+                $tokens->clearTokenAndMergeSurroundingWhitespace($testClose);
+
+                if (!$tokens[$testOpen + 1]->isWhitespace()) {
+                    $tokens->insertAt($testOpen + 1, new Token(array(T_WHITESPACE, ' ')));
+                }
+
+                if (false !== $testNsToken) {
+                    $tokens->clearTokenAndMergeSurroundingWhitespace($testNsToken);
+                }
+        }
+
+        return $assertCallClose;
+    }
+
+    /**
+     * @param Tokens    $tokens
+     * @param int|false $callNSIndex
+     * @param int       $callIndex
+     * @param int       $openIndex
+     * @param int       $closeIndex
+     */
+    private function removeFunctionCall(Tokens $tokens, $callNSIndex, $callIndex, $openIndex, $closeIndex)
+    {
+        $tokens->clearTokenAndMergeSurroundingWhitespace($callIndex);
+        if (false !== $callNSIndex) {
+            $tokens->clearTokenAndMergeSurroundingWhitespace($callNSIndex);
+        }
+
+        $tokens->clearTokenAndMergeSurroundingWhitespace($openIndex);
+        $tokens->clearTokenAndMergeSurroundingWhitespace($closeIndex);
+    }
+
+    /**
+     * @return array Token sequence
+     */
+    private function getSearchSequence()
+    {
+        static $sequence;
```

Comment:
can't it's be declared `static $foo = [];` ?
I can't see the need for lazy initialization here